File: python/src/unittest/python/GoogleAdapterTest.py

from GoogleDocument import *
from GoogleAdapter import *
from Format import *
from BackImage import *

import unittest


class GoogleAdapterTest(unittest.TestCase):

    def testInvalidLicense(self):
        self.assertFalse(self.word_doc.
                         licenceInvalidCannotEdit())

    def setUp(self):
        self.google_doc = GoogleDocument()
        self.word_doc = GoogleAdapter(self.google_doc)

    def testFormat(self):
        self.assertEquals(Font('Arial', 11).getFontText(),
                          self.word_doc.get_format()
                          .getFont().getFontText())


    def testBackground(self):
        self.assertEquals(BackgroundImage()
                          .getImage().get_url(),
                          self.word_doc
                          .getBackground().get_url())

    def testLicense(self):
        self.assertEquals(1,
                          self.word_doc.getOfficeLicense()
                          .getLicSerialNum())

File: java/src/main/java/Test/GoogleAdapterTest.java
package Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;

import mx.iteso.adapter.BackgroundImage;
import mx.iteso.adapter.googledocs.GoogleDoc;
import mx.iteso.adapter.googledocs.GoogleDocsAdapter;
import mx.iteso.adapter.word.IWordDocument;

import java.awt.*;

public class GoogleAdapterTest {
    private GoogleDoc googleDocs = new GoogleDoc();
    private IWordDocument wordDocument = new GoogleDocsAdapter(googleDocs);

    public void testFormat(){
        assertEquals(new Font("Roboto", Font.ROMAN_BASELINE, 25), this.wordDocument.getFormat().getFont());
    }

    public  void testBackground(){
        assertEquals(new BackgroundImage(this.wordDocument.getBackground()).getImage().getUrl(), this.wordDocument.getBackground().getUrl());
    }

    public void testLicense(){
        assertEquals(0, this.wordDocument.getLicense().getLicense());
    }

}
